aioredis/util.py

```diff
@@ -55,6 +55,62 @@ def decode(obj, encoding):
     return obj
 
 
+class Queue:
+
+    def __init__(self, connection, key=uuid.uuid4().hex, max_size=0, timeout=None):
+        self._conn = connection
+        self._key = key
+        self._max_size = max_size
+        self._timeout = timeout
+
+    @property
+    def key(self):
+        return self._key
+
+    @property
+    def max_size(self):
+        return self._max_size
+
+    @asyncio.coroutine
+    def qsize(self):
+        return (yield from self._conn.llen(self._key))
+
+    @asyncio.coroutine
+    def empty(self):
+        return (yield from self.qsize()) == 0
+
+    @asyncio.coroutine
+    def full(self):
+        if self._max_size != 0:
+            return (yield from self.qsize()) >= self._max_size
+        return False
+
+    @asyncio.coroutine
+    def get(self, timeout=None):
+        query_timeout = timeout if timeout else self._timeout
+        item = yield from self._conn.blpop(self._key, timeout=query_timeout)
+        return item
+
+    @asyncio.coroutine
+    def get_nowait(self):
+        item = yield from self._conn.lpop(self._key)
+        return item
+
+    @asyncio.coroutine
+    def put(self, data):
+        if not (yield from self.full()):
+            yield from self._conn.rpush(self._key, data)
+            return True
+        return False
+
+    @asyncio.coroutine
+    def put_nowait(self, data):
+        if not (yield from self.full()):
+            asyncio.ensure_future(self._conn.rpush(self._key, data))
+            return True
+        return False
```

Comment:
Thank you for the feedback.
Are you suggesting that it would be better to use lua script to implement this and wrap it with python using lupa?